hdfs-plugins/src/main/java/co/cask/hydrator/plugin/HDFSSink.java

```diff
@@ -122,18 +122,26 @@ private static String outputFormatClassName(String option) {
 
     @Name("path")
     @Description("HDFS Destination Path Prefix. For example, 'hdfs://mycluster.net:8020/output")
+    @Macro
     private String path;
 
     @Name("suffix")
     @Description("Time Suffix used for destination directory for each run. For example, 'YYYY-MM-dd-HH-mm'. " +
       "By default, no time suffix is used.")
     @Nullable
+    @Macro
     private String timeSufix;
 
     public HDFSSinkConfig(String referenceName, String path, String suffix, String outputFormat) {
       super(referenceName);
       this.path = path;
       this.timeSufix = suffix;
     }
+
+    private void validate() {
+      if (!containsMacro("timeSuffix") && !Strings.isNullOrEmpty(timeSufix)) {
```

Comment:
The check is used because at configure time, "timeSuffix" will contain macro syntax (e.g. `${timeSuffix}`) so we don't want to try and parse that as a valid date and defer the check until runtime when the field will not contain a macro.